kyo-core/shared/src/main/scala/kyo/StreamCoreExtensions.scala

```diff
@@ -1,32 +1,48 @@
 package kyo
 
+import kyo.Async.defaultConcurrency
 import kyo.kernel.ArrowEffect
 
 object StreamCoreExtensions:
-    val DefaultCollectBufferSize = 1024
+    val defaultAsyncStreamBufferSize = 1024
 
     private def emitMaybeChunksFromChannel[V](channel: Channel[Maybe[Chunk[V]]])(using Tag[V], Frame) =
         val emit = Loop(()): _ =>
             channel.take.map:
-                case Absent => Loop.done
+                case Absent =>
+                    Loop.done
                 case Present(c) =>
                     Emit.valueWith(c)(Loop.continue)
         Abort.run(emit).unit
     end emitMaybeChunksFromChannel
 
+    private def emitMaybeElementsFromChannel[V](channel: Channel[Maybe[V]])(using Tag[V], Frame) =
+        val emit = Loop(()): _ =>
+            channel.take.map: v =>
+                channel.drain.map: chunk =>
+                    val fullChunk      = Chunk(v).concat(chunk)
+                    val publishedChunk = fullChunk.collect({ case Present(v) => v })
```

Comment:
The transformation of chunks here isn't very efficient. What do you think about using a `ChunkBuilder?

Reply:
Given that this will only ever prepend `v` to a chunk once and is not building chunks by repeated concatenation, I suspect this is more efficent than using a `ChunkBuilder`, which would require iterating over the entire chunk that `v` is being prepended to.

Reply:
My concern is more the `collect` call since it'll use Scala's implementation, which can be expensive. I think it's fine to optimize it later, though

Reply:
@hearnadam is this a concern? At this point, doesn't `Chunk` use `ChunkBuilder` by default to transformations via `StrictOptimizedSeqOps`?

Reply:
hmm good point! makes sense

Reply:
If not I can add it to `Chunk` in a separate PR